sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -181,11 +195,111 @@ public SeekableByteChannel newByteChannel(Path path, Set<? extends OpenOption> s
     }
 
     /**
+     * The existence of a directory in the {@code AzureFileSystem} is defined on two levels. Weak existence is defined
+     * by the presence of a non-zero number of blobs prefixed with the directory's path. This is also known as a virtual
+     * directory and is in order to accommodate containers that were pre-loaded with data by another source but need to
+     * be accessed by this file system. Strong existence is defined as the presence of a zero-length blob whose name is
+     * the directory path with a particular metadata field indicating the blob's status as a directory and is also known
+     * as a concrete directory. Directories created by this file system will strongly exist. Operations targeting
+     * directories themselves as the object (e.g. setting properties) will target marker blobs. Operations using the
+     * directory only as a container (e.g. listing) will operate on the blob-name prefix.
+     *
+     * This method fulfills the contract of: "The check for the existence of the file and the creation of the directory
+     * if it does not exist are a single operation that is atomic with respect to all other filesystem activities that
+     * might affect the directory." The action of checking whether the parent exists, is not, however, atomic with
+     * the creation of the directory. While it is possible that the parent may be deleted between the positive check and
+     * the creation of the child, the creation of the child will always implicitly ensure the existence of a virtual
+     * parent, so the child will never be left floating and unreachable. Note that we only check for weak existence of
+     * the parent directory and only check for strong existence of an actual blob or other directory when attempting to
+     * put an object at the given path. This split is due to limitations in the Storage service API.
+     *
+     * There may be some unintuitive behavior when working with directories in this file system, particularly virtual
+     * directories(usually those not created by this file system). A virtual directory will disappear as soon as all its
+     * children have been deleted. Furthermore, if a directory with the given path weakly exists at the time of calling
+     * this method, this method will still return success and create a concrete directory at the target location.
+     * In other words, it is possible to "double create" a directory if it first weakly exists and then is strongly
+     * created. This is both because it is impossible to atomically check if a virtual directory exists while creating a
+     * concrete directory and because such behavior will have minimal side effects--no files will be overwritten and the
+     * directory will still be available for writing as intended, though it may not be empty.
```

Comment:
`may not` -> `will not`, as it definitely won't be empty if the directory virtually existed already.

Reply:
I used "may" because we don't know if we're creating a new directory or double creating a virtual directory. So there are cases of calling this method when the directory will be empty.